plugins/yaml/src/org/jetbrains/yaml/references/YAMLAliasReference.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2000-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jetbrains.yaml.references;
+
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiReferenceBase;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.ArrayUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.yaml.psi.YAMLScalar;
+
+import java.util.Collection;
+import java.util.Comparator;
+
+public class YAMLAliasReference extends PsiReferenceBase<PsiElement> {
+
+  private static final String ANCHOR_PREFIX = "&";
+
+  public YAMLAliasReference(@NotNull PsiElement element) {
+    super(element);
+  }
+
+  @Nullable
+  @Override
+  public PsiElement resolve() {
+    Collection<YAMLScalar> values = PsiTreeUtil.findChildrenOfType(myElement.getContainingFile(), YAMLScalar.class);
```

Comment:
@valich I played around a bit with JFlex, what a great tool! I added YAML anchor support in YAML flexer by modifying the `KEY_block` regex to this

```
KEY_block = {NS_PLAIN_ONE_LINE_block} {WHITE_SPACE_CHAR}* ":" (\ &[^\[\]\{\}\n\t\r\ ]*)?
```
Sample:
```
node1:
    node11: &node11
        node111: value111
        node112: value112
    node12: *node11
```

![image](https://cloud.githubusercontent.com/assets/1151536/26726840/de1659c8-47a4-11e7-95e4-6f0a8eba71d2.png)

The anchor is part of the `YAMLKeyValue`'s key. We could then add methods such as `isAnchorKey()` and `getAnchorName()` to `YAMLKeyValue`. 

Should we create a new class for the anchors and have a method like this (?): 

```
@Nullable
  public PsiElement getAnchor() {
    final PsiElement result = findChildByType(YAMLTokenTypes.ANCHOR);
    if (result != null) {
      return result;
    }
    return null;
  }

Or is it enough to have the anchor information in the key as String?